src/pages/EnvironmentsPage/environmentsPermissions.ts

```diff
@@ -0,0 +1,26 @@
+import { EnvironmentPermissionsProvider } from 'src/analysis/Environments/Environments';
+import { getCreatorForCompute } from 'src/analysis/utils/resource-utils';
+import { PersistentDisk } from 'src/libs/ajax/leonardo/models/disk-models';
+import { getTerraUser } from 'src/libs/state';
+
+const makePermissionsProvider = (userEmailGetter: () => string): EnvironmentPermissionsProvider => {
+  const permissionsProvider: EnvironmentPermissionsProvider = {
+    canDeleteDisk: (disk: PersistentDisk) => {
+      const currentUserEmail = userEmailGetter();
+      const canDelete = disk.auditInfo.creator === currentUserEmail;
+      return canDelete;
+    },
+    canPauseResource: (resource) => {
+      const currentUserEmail = userEmailGetter();
+      const creator = getCreatorForCompute(resource);
+      return currentUserEmail === creator;
+    },
+  };
+  return permissionsProvider;
+};
+
+const currentUserEmailGetter = (): string => {
+  return getTerraUser().email!;
```

Comment:
It looks like this relies on Terra state, will this need to be abstracted away.

Is this one of the "Terra-specific" dependencies you mentioned on Slack?

Reply:
Never mind. I see now that this is the provider used by Terra where AoU would provide its own.

Reply:
correct :)